webapi/src/main/scala/org/knora/webapi/responders/v1/ValuesResponderV1.scala

```diff
@@ -345,6 +349,130 @@ class ValuesResponderV1 extends ResponderV1 {
     }
 
     /**
+      * Temporary structure to represent existing file values of a resource.
+      *
+      * @param property       the property Iri (e.g., hasStillImageFileValueRepresentation)
+      * @param valueObjectIri the Iri of the value object.
+      * @param quality        the quality of the file value
+      */
+    private case class CurrentFileValue(property: IRI, valueObjectIri: IRI, quality: Option[Int])
+
+    private def changeFileValueV1(changeFileValueRequest: ChangeFileValueRequestV1): Future[ChangeFileValueResponseV1] = {
+
+        // get the Iris of the current file value(s)
+        val resultFuture = for {
+
+            resourceIri <- Future(changeFileValueRequest.resourceIri)
+
+            getFileValuesSparql = queries.sparql.v1.txt.getFileValuesForResource(resourceIri = resourceIri).toString()
+            //_ = print(getFileValuesSparql)
+            getFileValuesResponse: SparqlSelectResponse <- (storeManager ? SparqlSelectRequest(getFileValuesSparql)).mapTo[SparqlSelectResponse]
+            // _ <- Future(println(getFileValuesResponse))
+
+
+            // get the property Iris, file value Iris and qualities attached to the resource
+            fileValues: Seq[CurrentFileValue] = getFileValuesResponse.results.bindings.map {
+                (row: VariableResultsRow) =>
+
+                    CurrentFileValue(
+                        property = row.rowMap("p"),
+                        valueObjectIri = row.rowMap("fileValueIri"),
+                        quality = row.rowMap.get("quality") match {
+                            case Some(quality: String) => Some(quality.toInt)
+                            case None => None
+                        }
+                    )
+            }
+
+            sipiConversionMessage: SipiResponderConversionRequestV1 = changeFileValueRequest.file
+
+            // if resource has no existing file values, throw an exception
+            _ = if (fileValues.isEmpty) {
+                BadRequestException(s"File values for ${resourceIri} should be updated, but resource has no existing file values")
+            }
+
+            // the message to be sent to Sipi responder
+            sipiConversionRequest: SipiResponderConversionRequestV1 = changeFileValueRequest.file
+
+            sipiResponse: SipiResponderConversionResponseV1 <- (responderManager ? sipiConversionRequest).mapTo[SipiResponderConversionResponseV1]
+
+            // check if the file type returned by Sipi corresponds to the already existing file value type (e.g., hasStillImageRepresentation)
+            _ = if (SipiConstants.fileType2FileValueProperty(sipiResponse.file_type) != fileValues.head.property) {
+                // TODO: remove the file from SIPI (delete request)
```

Comment:
Why is this TODO still here? I thought this was taken care of by exception handling.